tests/NLog.UnitTests/Internal/StringBuilderExtTests.cs

```diff
@@ -74,5 +75,58 @@ void TestAppendInvariant(int input)
             StringBuilderExt.AppendInvariant(sb, input);
             Assert.Equal(input.ToString(System.Globalization.CultureInfo.InvariantCulture), sb.ToString());
         }
+
+        [Theory]
+        [MemberData(nameof(TestAppendXsdDateTimeRoundTripCases))]
+        void TestAppendXsdDateTimeRoundTrip(DateTime input)
+        {
+            StringBuilder sb = new StringBuilder();
+            StringBuilderExt.AppendXsdDateTimeRoundTrip(sb, input);
+            Assert.Equal(System.Xml.XmlConvert.ToString(input, System.Xml.XmlDateTimeSerializationMode.Utc), sb.ToString());
+
+            input = new DateTime(input.Ticks, DateTimeKind.Local);
+            sb.Length = 0;
+            StringBuilderExt.AppendXsdDateTimeRoundTrip(sb, input);
+            Assert.Equal(System.Xml.XmlConvert.ToString(input, System.Xml.XmlDateTimeSerializationMode.Utc), sb.ToString());
+
+            input = new DateTime(input.Ticks, DateTimeKind.Utc);
+            sb.Length = 0;
+            StringBuilderExt.AppendXsdDateTimeRoundTrip(sb, input);
+            Assert.Equal(System.Xml.XmlConvert.ToString(input, System.Xml.XmlDateTimeSerializationMode.Utc), sb.ToString());
+        }
+
+        public static IEnumerable<object[]> TestAppendXsdDateTimeRoundTripCases()
+        {
+            yield return new object[] { DateTime.MinValue };
+            yield return new object[] { DateTime.MaxValue };
+            yield return new object[] { new DateTime(123, 1, 2) };
+            yield return new object[] { new DateTime(1234, 10, 20) };
+            yield return new object[] { new DateTime(1970, 1, 1, 1, 1, 1) };
+            yield return new object[] { new DateTime(1970, 1, 1, 1, 1, 1, 1) };
+            yield return new object[] { new DateTime(1970, 10, 10, 10, 10, 10) };
```

Comment:
I think the values should be unique, e.g. not same day and month, otherwise you could switch those without triggering a test. 

(Maybe that's covered with addticks, but the addticks cases looks a bit  arbitrary and could be removed  (in the future))

Reply:
The most exciting part is the removal of trailing zeros in ticks. So I like the AddTicks-cases.